backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/BasicScheduleGenerator.java

```diff
@@ -0,0 +1,183 @@
+    public Schedule generate() {
+        return new Schedule(description, availability, generateTrips());
+    }
+
+    private List<Trip> generateTrips() {
+        val timePoints = generateTimePoints();
+        timePoints.sort(Comparator.comparing(TimePoint::time));
+
+        val trips = new ArrayList<Trip>();
+        for (int vehId = 1; hasNotServicedTimePoints(timePoints); vehId++) {
+            TimePoint startPoint = getNextNotServicedTimePoint(timePoints);
+            trips.addAll(generateVehicleTrips(vehId, startPoint, timePoints));
+        }
+
+        return trips;
+    }
+
+    private List<TimePoint> generateTimePoints() {
+        val timePoints = new ArrayList<TimePoint>();
+        LocalTime nextTimePoint = startTimeInbound;
+        while (nextTimePoint.isBefore(endTimeInbound)) {
+            timePoints.add(new TimePoint(INBOUND, nextTimePoint, false));
+            nextTimePoint = nextTimePoint.plus(headway);
+        }
+        nextTimePoint = startTimeOutbound;
+        while (nextTimePoint.isBefore(endTimeOutbound)) {
+            timePoints.add(new TimePoint(OUTBOUND, nextTimePoint, false));
+            nextTimePoint = nextTimePoint.plus(headway);
+        }
+        return timePoints;
+    }
+
+    private List<Trip> generateVehicleTrips(int vehicleId, TimePoint startPoint, List<TimePoint> timePoints) {
+        val trips = new ArrayList<Trip>();
+
+        while (true) {
+            val trip = generateTrip(vehicleId, startPoint);
+            trips.add(trip);
+
+            //TODO maybe remove last trip and break if its arrival time is after end time? May be some parameter to indicate possible lateness?
+            val currentTime = getArrivalTime(trip);
+            val currentDirection = INBOUND.equals(startPoint.direction()) ? OUTBOUND : INBOUND;
+            val nextPointOpt = findNextNotServicedTimePointAfter(timePoints, currentTime, currentDirection);
+
+            if (nextPointOpt.isPresent()) {
+                startPoint = nextPointOpt.get();
+            } else {
+                break;
+            }
+        }
+
+        return trips;
+    }
+
+    private Trip generateTrip(int vehicleId, TimePoint timePoint) {
+        timePoint.isServiced(true);
+        val route = INBOUND.equals(timePoint.direction()) ? routeInbound : routeOutbound;
+        return new Trip(
+            route,
+            vehicleId,
+            route.stations().get(route.stations().size() - 1).name(),
+            calculateStops(timePoint.time(), route, averageSpeed, dwellTime));
+    }
+
+    private List<Stop> calculateStops(LocalTime startTime, Route route, Quantity<Speed> averageSpeed, Duration dwellTime) {
+        val stops = new ArrayList<Stop>();
+
+        val speedValue = averageSpeed.to(Units.METRE_PER_SECOND).getValue().longValue();
+
+        for (int i = 0; i < route.stations().size(); i++) {
+            val station = route.stations().get(i);
+            val distTravelled = route.shape().shapePoints().stream()
```

Comment:
Consider extracting to another method like `Optional<Quantity<Length>> distanceTraveled(Route r)`